pkg/containerd/cmd/service/start.go

```diff
@@ -0,0 +1,123 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/containerd/containerd"
+	"github.com/containerd/containerd/api/services/containers"
+	"github.com/containerd/containerd/api/services/execution"
+	//"github.com/containerd/containerd/api/types/mount"
+	"github.com/containerd/containerd/namespaces"
+	specs "github.com/opencontainers/runtime-spec/specs-go"
+)
+
+func start(args []string) {
+	invoked := filepath.Base(os.Args[0])
+	flags := flag.NewFlagSet("start", flag.ExitOnError)
+	flags.Usage = func() {
+		fmt.Printf("USAGE: %s start [service]\n\n", invoked)
+		fmt.Printf("Options:\n")
+		flags.PrintDefaults()
+	}
+
+	sock := flags.String("sock", "/run/containerd/containerd.sock", "Path to containerd socket")
+
+	dumpSpec := flags.String("dump-spec", "", "Dump container spec to file before start")
+
+	if err := flags.Parse(args); err != nil {
+		log.Fatal("Unable to parse args")
+	}
+	args = flags.Args()
+
+	if len(args) != 1 {
+		fmt.Println("Please specify the service")
+		flags.Usage()
+		os.Exit(1)
+	}
+
+	service := args[0]
+	rootfs := filepath.Join("/containers/services", service, "rootfs")
+	log.Infof("Starting service: %q", service)
+	log := log.WithFields(log.Fields{
+		"service": service,
+	})
+
+	client, err := containerd.New(*sock)
+	if err != nil {
+		log.Fatal("creating containerd client")
+	}
+
+	ctx := namespaces.WithNamespace(context.Background(), "default")
+
+	var spec *specs.Spec
+	specf, err := os.Open(filepath.Join("/containers/services", service, "config.json"))
+	if err != nil {
+		log.Fatalf("failed to read config.json: %s", err)
```

Comment:
I stand corrected, I made `stdin` be `/dev/nullx` and with both `%s` and `%v` the output is:
```
FATA[0000] failed to create task: rpc error: code = NotFound desc = stat /dev/nullx: no such file or directory  service=getty
```
Closer reading of https://golang.org/pkg/fmt/ confirms that both `%v` and `%s` will, if the argument implements the `Error` interface, call the `Error()` method (which is what I meant when I said `String()` above):
>  If the format (which is implicitly %v for Println etc.) is valid for a string (%s %q %v %x %X), the following two rules apply:
>
> 4. If an operand implements the error interface, the Error method will be invoked to convert the object to a string, which will then be formatted as required by the verb (if any).
>
> 5. If an operand implements method String() string, that method will be invoked to convert the object to a string, which will then be formatted as required by the verb (if any). 

I was sure I had seen garbled and unhelpful error messages from the use of `%v` on error types in the past, I must have been confused.